Update notifications and info boxes styles for accessibility

bx80 opened the pull request:
### Description:

Fixes #20691

The PR updates the styling of notifications and info boxes throughout the application to improve accessibility.

For some reason we have components for both notifications and alerts, each of which can be of the types: success, info, warning or error. The only real difference I can see is that notifications can optionally have a close button. There seems to be roughly equal usage of both components throughout the application. If there is no other difference then we might want consider deprecating alerts at some point in favor of notifications.

I’ve applied the new styles to both notifications and alerts and also added some notification examples to the UI demo screen so that they can be covered by UI tests.

### Review

* [ ] [Functional review done](https://developer.matomo.org/guides/pull-request-reviews#functional-review-done)
* [ ] [Potential edge cases thought about](https://developer.matomo.org/guides/pull-request-reviews#potential-edge-cases-thought-about) (behavior of the code with strange input, with strange internal state or possible interactions with other Matomo subsystems)
* [ ] [Usability review done](https://developer.matomo.org/guides/pull-request-reviews#usability-review-done) (is anything maybe unclear or think about anything that would cause people to reach out to support)
* [ ] [Security review done](https://developer.matomo.org/guides/security-in-piwik#checklist)
* [ ] [Wording review done](https://developer.matomo.org/guides/pull-request-reviews#translations-wording-review-done)
* [ ] [Code review done](https://developer.matomo.org/guides/pull-request-reviews#code-review-done)
* [ ] [Tests were added if useful/possible](https://developer.matomo.org/guides/pull-request-reviews#tests-were-added-if-usefulpossible)
* [ ] [Reviewed for breaking changes](https://developer.matomo.org/guides/pull-request-reviews#reviewed-for-breaking-changes)
* [ ] [Developer changelog updated if needed](https://developer.matomo.org/guides/pull-request-reviews#developer-changelog-updated-if-needed)
* [ ] [Documentation added if needed](https://developer.matomo.org/guides/pull-request-reviews#documentation-added-if-needed)
* [ ] Existing documentation updated if needed

Commit:
Update notifications, alerts and info boxes to new design

Commit:
Update notifications, alerts and info boxes to new design

Commit:
Updated demo notifications to add multiline example and comment on close button.

michalkleiner commented:
Created new ticket to look into the consolidation of the components — https://github.com/matomo-org/matomo/issues/21079.

Commit:
Replace 89 matomo font glyphs with new icons, consolidated icon-info2 usage to icon-info, added source SVG files for new icons

Commit:
Replace 89 matomo font glyphs with new icons, consolidated icon-info2 usage to icon-info, added source SVG files for new icons

bx80 committed:
built vue files

Commit:
Bump stalled tests

Commit:
Bump stalled tests

Commit:
Updated Matomo font icons scale, tweak for export icon css, updated reporting actions icon

Commit:
Updated Matomo font icons scale, tweak for export icon css, updated reporting actions icon

Commit:
Updated UI test screenshots for icon and notification changes

Commit:
Updated UI test screenshots for icon and notification changes

Commit:
Updated UI test screenshots after rebase

sgiehl commented:
@bx80 for such intense changes it's good to temporarily update the submodules to the correlating branches in the PR. That way you can directly see if all tests of the submodule plugins are passing

sgiehl commented on plugins/Login/tests/UI/expected-screenshots/Login_bruteforcelog_noentries.png:
Looks like the icon for the platform menu got lost.

bx80 replied:
I consolidated icon-user-platform and icon-admin-platform as they were the same, that should be fixed now.

bx80 replied:
I've confirmed this is fixed on the latest tests.

sgiehl commented on plugins/Marketplace/tests/UI/expected-screenshots/Marketplace_notification_plugincheck_exceededLicense.png:
Should links be blue in those notifications?

bx80 replied:
The notification mockups have all the links in black.
@Javi-Ormaechea can I confirm that the notification links should be the same colour as the rest of the notification text or do we want to have them in blue to match the rest of the application?

sgiehl commented on plugins/Marketplace/tests/UI/expected-screenshots/Marketplace_paid_plugin_details_exceeded_license_superuser.png:
Guess we missed that when changing the links, but guess those for authors and website should be blue, too.

bx80 replied:
I've updated the marketplace .metadata CSS class to make them blue.

sgiehl commented on plugins/Marketplace/tests/UI/expected-screenshots/Marketplace_superuser_enable_plugins_admin_with_multiserver_enabled.png:
The more links could be blue here.

bx80 replied:
Should be fixed now.

sgiehl commented on plugins/Morpheus/Controller.php:
```diff
@@ -96,13 +97,53 @@ public function demo()
+      <strong>Error</strong><br>
+      Change <a>a few things</a> and try submitting again, this notification does not have a close button, but it does contain multiple lines of text. Textus sine sensu ultra finem lineae pergit ante movens infra et iterum ad sinistram.
+    </Notification>',
+            'desc' => 'This is a Vue component. You can use the :noclear="true" property to hide the close button',
```
So is the last remaining difference between Alert and Notification, that Notifications by default are "clearable"?
Or is there anything else? If not we could maybe consider to deprecate/remove one of them in the future.

bx80 replied:
As far as I can see that's the only remaining difference, @michalkleiner has already created a ticket to deprecated one in favour of the other :+1:

sgiehl commented on plugins/Overlay/tests/UI/expected-screenshots/Overlay_loaded_from_actions.png:
The notification here looks cut off. We should check the reason maybe.

bx80 replied:
That particular test is taking a screenshot of a popup which is defaulting to 800x600, all the rest of the screenshots are 1350x768. I can't see any obvious way to resize the popup in the UI test but I don't think it's related to this change since the previous screenshot is also cut off.

sgiehl commented on plugins/PrivacyManager/tests/UI/expected-screenshots/PrivacyManager_anonymizelogdata_anonymizeip_and_visit_column_cancelled.png:
Shouldn't all drop downs use the same arrow? Our customized ones now have another arrow then the default drop downs rendered by materialize. We can go with that for sure, it just looks a bit inconsistent maybe.

bx80 replied:
They certainly should be consistent :+1: There is an additional ticket #20475 created to unify the drop down design (which I'll work on as soon as this PR is complete)

sgiehl commented on plugins/PrivacyManager/tests/UI/expected-screenshots/PrivacyManager_users_opt_out_default.png:
The white font on the light blue background might be a bit hard to read 🙈

bx80 replied:
Ouch, I was responsible for that monstrosity! I've removed the text colour override and it's now a standard readable info notification. :sunglasses:

bx80 replied:
Confirmed fixed in the latest test run.

sgiehl commented on plugins/SegmentEditor/tests/UI/expected-screenshots/SegmentSelectorEditorTest_updated.png:
That one doesn't have the correct result anymore. It should show the new name of the segment. Maybe the screenshot needs another update.

bx80 replied:
I've rebased this PR on 5.x-dev, so hopefully that will fix it on the next run.

sgiehl commented on plugins/SegmentEditor/tests/UI/expected-screenshots/SegmentSelectorEditorTest_updated_details.png:
Needs an update as well, as the segment name isn't correct anymore.

sgiehl commented on plugins/TwoFactorAuth/tests/UI/expected-screenshots/TwoFactorAuthUsersManager_edit_with_2fa_reset_confirmed.png:
The back link here uses `icon-arrow-left-2`, which now looks almost the same as the new `icon-arrow-left`. Maybe we should remove the second one, as it's only used in the usersmanage anyway.

bx80 replied:
I've moved `icon-arrow-left-2` and `icon-arrow-right-2` to the deprecated section of the font selection and updated user manager to use `icon-arrow-left`. Just trying to be cautious with deleting them entirely just yet in case there is a plugin somewhere using them.

sgiehl commented on plugins/UserCountryMap/tests/UI/expected-screenshots/VisitorMap_avg_time_on_site.png:
The tooltip is missing, which should be part of the UI tests. If the test is broken in general, we should create a new issue to fix it. But we should not update the screenshot here in case it doesn't have the correct result.

bx80 replied:
I'll revert this in the next commit :+1:

sgiehl commented on plugins/UserCountryMap/tests/UI/expected-screenshots/VisitorMap_bounce_rate.png:
Same here, tooltip is missing.

bx80 replied:
I'll revert this in the next commit :+1:

sgiehl commented on plugins/UsersManager/tests/UI/expected-screenshots/UsersManager_permissions_bulk_access_set.png:
That update is also incorrect. The back link is missing. Guess this was one of the random failures...

bx80 replied:
I've seen that one a few times, I think it's a scrolling thing, will revert.

sgiehl commented on plugins/UsersManager/tests/UI/expected-screenshots/UsersManager_superuser_tab.png:
Note: The new icon might look a bit less important in this case than before.

bx80 replied:
It's a bit small and not being a solid colour it's harder to see. I've slightly increased the size of the icon just for this usage. 
![image](https://github.com/matomo-org/matomo/assets/88810029/b8f961c7-4d64-4bc9-891d-150d05481617)

bx80 replied:
Confirmed the larger icon is now shown on the latest tests.

sgiehl commented on tests/UI/expected-screenshots/UIIntegrationTest_actions_pages_tooltip_help.png:
Should we maybe change this help box to look the same as an info notification?

bx80 replied:
The bottom right italic text might not work well in an info notification. Maybe there are other improvements that could be done to the report info help box at the same time so it might be worth getting some UX input on another ticket.

sgiehl commented on tests/UI/expected-screenshots/UIIntegrationTest_category_help.png:
This doesn't look better then before. Maybe we are missing some definitions how that help notification should look like?

bx80 replied:
Yes that looks awful, help notifications didn't get covered in the mockups, the style is a derivative of the info notification. For now I've removed the dark blue background from the style but retained the different icon and it looks a lot better. @Javi-Ormaechea can review when the next round of UI tests have completed.

![image](https://github.com/matomo-org/matomo/assets/88810029/76869598-78c0-4dd9-9d99-afdaaa6401e3)

bx80 replied:
I can confirm this is showing the updated info style on the latest test screenshots.

sgiehl reviewed (changes requested):
Had a look through all the changed screenshots. Left a couple of comments, where additional changes might be required.
Some of them might be unrelated to the purpose of this PR and could also be changed independently. In some cases the resulting screenshots aren't correct anymore and need further updates.

In general we should maybe discuss if we want to deprecate either the Notification or the Alert component. It seems they already look the same, so their is no benefit in having both.

Commit:
Various notification, style and icon improvements

- Temporarily updated submodule references for testing
- Updated the 25 outstanding icons used with core, both with .svg files and updated font files
- icon-done now uses the same thin tick icon as icon-ok
- icon-plus-square and icon-minus-square now have squares again
- New icons icon-add1 and icon-remove have added
- New icons icon-chevron-right and icon-chevron-left have been added
- Updated mobile down menu, dashboard settings, site selector, fieldExandableSelect and reporting menu to use a chevron as before
- The Morpheus UI demo icons section has been tidied to only show the new icon set
- Fixed link colour on marketplace metadata sidebar
- Fixed colour on marketplace plugin tile more link
- Fixed privacy manager opt out non-standard notification
- Increased the size of the warning icon on user permissions menu
- Adjusted help notification background colour
- Reverted three UI tests from a previous commit which were not related to style changes

bx80 commented:
I've committed the following changes which should address all issues raised so far:

- Temporarily updated submodule references for easier testing :warning: MUST REVERT BEFORE MERGING :warning:
- Updated the 25 outstanding icons used within core, both individual .svg files and updated font files.
- `icon-done` now uses the same thin tick icon as `icon-ok`.
- `icon-plus-square` and `icon-minus-square` now have squares again.
- New icons `icon-add1` and `icon-remove` have been added for where we need simple + / -.
- New icons `icon-chevron-right` and `icon-chevron-left` have been added.
- Updated mobile down menu, dashboard settings, site selector, fieldExandableSelect and reporting menu controls to use a chevron down arrows as before (this will be tidied up further in [#20475](https://github.com/matomo-org/matomo/issues/20475)
- The Morpheus UI demo icons section has been tidied to only show icons with the new style and some missing icons added.
- Fixed the link colour on the marketplace metadata sidebar.
- Fixed the colour on the marketplace plugin tile more link.
- Fixed the ugly privacy manager opt-out non-standard notification colours.
- Increased the size of the warning icon on user permissions menu so it's more visible with outline icons.
- Adjusted the help notification background colour so it matches an info notification, just with a different icon.
- Reverted three UI test screenshot from a previous commit which were not related to style changes.
- Rebased on 5.x-dev

bx80 committed:
built vue files

Commit:
Bump stalled tests again

Commit:
Merge remote-tracking branch 'origin/m20691-notification-styles' into m20691-notification-styles

sgiehl commented on plugins/Marketplace/stylesheets/marketplace.less:
```diff
@@ -98,7 +98,7 @@
             margin-bottom: 10px;
             .more {
-                color: @theme-color-text;
+                color: #1976d2;
```
Should be better to use the variable.
```suggestion
                color: @theme-color-link;
```

sgiehl commented on plugins/Marketplace/stylesheets/marketplace.less:
```diff
@@ -111,6 +111,11 @@
             margin: 15px 15px 10px;
             list-style: none;
+            a {
+                color: #1976d2;
```
```suggestion
                color: @theme-color-link;
```

Commit:
Updated font with scaling tweaks

Commit:
Update UI screenshots after font change

sgiehl commented on plugins/Dashboard/tests/UI/expected-screenshots/DashboardManager_create_new.png:
The placement of the dropdown icons when displayed right looks not nice. See the dashboard selector in main menu

bx80 replied:
It seems okay to me, is this what you mean ? 
![image](https://github.com/matomo-org/matomo/assets/88810029/2c41ee76-f260-448e-9bff-7636be0db2bf)
There is another ticket to be done right after this one where dropdown alignment and icons will be tidied up, so maybe any tweaks to this should be done there?

sgiehl commented on plugins/Live/tests/UI/expected-screenshots/Live_visitor_profile_action_tooltip.png:
This one was incorrectly updated

sgiehl commented on plugins/Login/tests/UI/expected-screenshots/Login_password_reset_complete.png:
Might not be needed for this PR, but shouldn't this notification message maybe also use one of the new styles? Maybe the success style would match here, as it's shown after the password was successfully resetted?

bx80 replied:
The same infoMessage box is also used to notify that the password link has been sent, so I've updated it to use the info style since that will work for both.

bx80 replied:
I can confirm the latest test screenshots now show use the info style

sgiehl commented on plugins/SegmentEditor/tests/UI/expected-screenshots/SegmentSelectorEditorTest_autocomplete_capitalized.png:
this one has incorrect changed content

bx80 replied:
The segment selector tests are broken on other [builds](https://builds-artifacts.matomo.org/matomo-org/matomo/DEV-16595-no-data-screen-translations-review/5793646777/) too, so it look like either something has recently changed in the test or it is randomly failing.

sgiehl commented on plugins/SegmentEditor/tests/UI/expected-screenshots/SegmentSelectorEditorTest_autocomplete_lowercase.png:
this one, too

sgiehl commented on plugins/SegmentEditor/tests/UI/expected-screenshots/SegmentSelectorEditorTest_autocomplete_uppercase.png:
and here...

sgiehl commented on plugins/SegmentEditor/tests/UI/expected-screenshots/SegmentSelectorEditorTest_updated.png:
also incorrectly updated

sgiehl commented on plugins/SegmentEditor/tests/UI/expected-screenshots/SegmentSelectorEditorTest_updated_details.png:
here as well

sgiehl commented on plugins/UserCountryMap/tests/UI/expected-screenshots/VisitorMap_avg_time_on_site.png:
Tooltip is now missing

bx80 replied:
Should now be fixed.

sgiehl commented on plugins/UserCountryMap/tests/UI/expected-screenshots/VisitorMap_bounce_rate.png:
Tooltip missing

bx80 replied:
Should now be fixed.

sgiehl commented on tests/UI/expected-screenshots/Menus_admin_loaded.png:
platform icon is still missing here

bx80 replied:
Looks like the previous fix got lost in the last font update somehow, it should be sorted now.

bx80 replied:
Confirmed fixed in the latest tests.

sgiehl reviewed (changes requested):
Had another look through the screenshots. Added a couple of comments, where screenshot changed the content unexpected, or some stuff wasn't correct.

bx80 committed:
Update plugins/Marketplace/stylesheets/marketplace.less

Co-authored-by: Stefan Giehl <stefan@matomo.org>

bx80 committed:
Update plugins/Marketplace/stylesheets/marketplace.less

Co-authored-by: Stefan Giehl <stefan@matomo.org>

Commit:
Fix incorrect UI test screenshots for visitor map and profile action tooltip

Commit:
Fix missing admin menu platform icon

Commit:
Password changed confirmation style updated to standard info box

Commit:
Update submodule references

Commit:
Update UI test screenshots

sgiehl committed:
update submodules

sgiehl committed:
updates expected UI test files

sgiehl committed:
update submodules

sgiehl committed:
update submodules

sgiehl committed:
use correct menu item

sgiehl committed:
submodule updates

sgiehl committed:
built vue files

sgiehl committed:
updates expected UI test files

sgiehl committed:
improve menu icon
